### PR TITLE
Add `logging::log` & `assert::assert_eq` to `prelude.sw`

### LINKED ISSUE #3914: Proposal: Add `logging::log` & `assert::assert_eq` to prelude

### PR DESCRIPTION
**Pull request by Braqzen**
## Changes

- Added `log` & `assert_eq` to `prelude.sw` and to docs
- Removed `prelude` imports from files
- Changed logging doc to explicitly state that it only logs "arbitrary `stack` types"
- Minor manual formatting here and there
- Updated `LSP` test since it relies on hardcoded values in a file which had an import that no longer exists (and thus the number of lines is different)

## Related

Closes #3914

### CODE REVIEW

**Review by nfurfaro — DISMISSED** [review]

File: docs/book/src/introduction/standard_library.md [hunk 1]
```diff
@@ -34,5 +34,7 @@ The current version of the prelude lives in [`std::prelude`](https://github.com/
 - [`std::option::Option`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/option.sw), an enum which expresses the presence or absence of a value.
 - [`std::result::Result`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/result.sw), an enum for functions that may succeed or fail.
 - [`std::assert::assert`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/assert.sw), a function that reverts the VM if the condition provided to it is `false`.
+- [`std::assert::assert_eq`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/assert.sw), a function that reverts the VM and logs if the condition `v1` == `v2` is `false`.
```
Comment (mohammadfawaz) [inline docs/book/src/introduction/standard_library.md:37]:
```suggestion
- [`std::assert::assert_eq`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/assert.sw), a function that reverts the VM and logs its two inputs `v1` and `v2` if the condition `v1` == `v2` is `false`.
```

File: docs/book/src/introduction/standard_library.md [hunk 2]
```diff
@@ -34,5 +34,7 @@ The current version of the prelude lives in [`std::prelude`](https://github.com/
...
 - [`std::revert::require`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/revert.sw), a function that reverts the VM and logs a given value if the condition provided to it is `false`.
 - [`std::revert::revert`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/revert.sw), a function that reverts the VM.
+- [`std::logging::log`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/logging.sw), a function that allows logging of arbitrary types.
```
Comment (mohammadfawaz) [inline docs/book/src/introduction/standard_library.md:40]:
```suggestion
- [`std::logging::log`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/logging.sw), a function that logs arbitrary stack types.
```

**Review by mohammadfawaz — COMMENTED** [review]
Just some nits.

**Review by mohammadfawaz — APPROVED** [review]
Great! Thank you for doing this.

**Review by nfurfaro — APPROVED** [review]

**Pull request merged**